### DISCUSSION TITLE
How to go back twice or many times with Next/Router

### DISCUSSION THREAD

**Discussion by yyhan0915** [vercel/next.js#18333] [Ideas]
As I checked in github source code, unlike react-router, 
next-router does not have multiple times of go back on **Router.go()** function.

But in some cases, such as routing back to previous page from error, it is needed to make code of multi times of go back.
So my idea is for doing it,

**Router.go(times : number) => window.window.history.go(times);**
 
How about my idea? Is there some feature I described above in NextJs?

**Comment by 1finedev** [comment]
any update? i need this

**Comment by shahidul5** [comment]
any update?

**Comment by shaianest** [comment]
window.history.go(delta) for example window.history.go(-3) goes back 3 times which is good
 
window is used and wont give any  error because you go back or forward in useEffect or Event callBacks.

**Comment by captain-yossarian** [comment]
Is there any chance it will be implemented ? It is useless to be able to go back only for 1 step.

**Comment by hamza-mashhadi** [comment]
facing an issue in trying to create an error page and go back more than 1 step by using router.back(), if you go only 1 back it makes an infinite loop between an error page and a 404 page. 
Sadly had to use @shaianest 's solution for now. 

any plans to integrate this anytime soon?

**Comment by Ubinquitous** [comment]
I think this is work.
```jsx
// window.history.go(-2)

router.back()
router.back()
```

**Reply by jay-herrera** [reply]
I've noticed a difference in behavior on iOS, where calling history.back() multiple times does not go back multiple times, whereas router.go(-n) does

**Comment by josurene** [comment]
Any update?
`packages/next/src/shared/lib/router/router.ts`
like, just modify the back function?
` 
  back(n = 1) {
    window.history.go(n)
  }
`

**Comment by alirezamehrabi** [comment]
just do it with javaScript, i could not find any way in next.js.

```
onClick={() => window.history.go(-1)}
```

**Reply by Rehaeel** [reply]
The nextjs navigation via `router` is shallow, but `window.history.go()` is not, thus there will be a refresh of the page.
